### PR TITLE
Check if ECC is enabled, and if so, disable it

### PR DESCRIPTION
**Pull request by ilkhom19**
## Summary by CodeRabbit

- **New Features**
  - Enhanced the setup script to check and manage the Error Correcting Code (ECC) status of NVIDIA GPUs.
  - Added functionality to report the number of GPUs detected in the system.
  - Implemented user notifications for ECC status changes and the necessity for a system reboot.

- **Bug Fixes**
  - Improved error handling and reporting for ECC configurations.

### CODE REVIEW

File: ionet-setup.sh [hunk 1]
```diff
@@ -46,6 +46,35 @@ else
...
+            # Read current ECC mode to decide if a change is necessary
+            current_ecc_mode=$(nvidia-smi -i $gpu_index --query-gpu=ecc.mode.current --format=csv,noheader,nounits)
+
+            if [ "$current_ecc_mode" = "Enabled" ]; then
+                echo "ECC is enabled on GPU ${gpu_index}, attempting to disable..."
+
+                # Set ECC mode to 0 (disabled) and reboot the GPU to apply
+                sudo nvidia-smi -i $gpu_index --ecc-config=0
+
+                echo "ECC has been disabled for GPU ${gpu_index}, a reboot may be required to apply changes."
```
Comment (vnlitvinov) [inline ionet-setup.sh:70]:
```suggestion
                echo "ECC has been disabled for GPU ${gpu_index}, system reboot is required to apply changes."
```

**Review by vnlitvinov — DISMISSED** [review]
Overall looks good, however bot comment on simplification makes sense, I suggest to take it

File: ionet-setup.sh [hunk 2]
```diff
@@ -46,6 +46,35 @@ else
             echo "CUDA version $cuda_version is installed but does not meet the minimum requirement of $min_version. Please upgrade CUDA."
             exit 1
         fi
+
+        # Check if ECC is enabled, and if so, disable it.
+        # Query the number of GPUs in the system
+        num_gpus=$(nvidia-smi --list-gpus | wc -l)
+
+        echo "Found $num_gpus GPUs in the system."
```
[the remark this answers is not in the document]
Comment (vnlitvinov) [reply ionet-setup.sh:54]: This doesn't make sense, we shouldn't quit here as we have other things to do yet

File: ionet-setup.sh [hunk 3]
```diff
@@ -58,22 +58,19 @@ else
...
+                else
+                    echo "Failed to disable ECC on GPU ${gpu_index}."
+                fi
             else
                 echo "ECC is already disabled on GPU ${gpu_index}."
             fi
         done
 
-        echo "ECC check/update completed. System may need a reboot to finalize changes."
+        echo "ECC check/update completed. A Reboot is required to finalize changes."
```
Comment (vnlitvinov) [inline ionet-setup.sh:73]: this should (ideally) be shown only when we did some changes

**Review by vnlitvinov — APPROVED** [review]

File: ionet-setup.sh (hunk 3 above)
Comment (ilkhom19) [reply ionet-setup.sh]: removed that line since we already prompt while doing action

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**